rllib/algorithms/marwil/marwil_torch_policy.py

```diff
@@ -69,27 +69,32 @@ def loss(
             adv_squared_mean = torch.mean(torch.pow(adv, 2.0))
 
             explained_var = explained_variance(cumulative_rewards, state_values)
-            self.explained_variance = torch.mean(explained_var)
+            ev = torch.mean(explained_var)
+            model.tower_stats["explained_variance"] = ev
 
             # Policy loss.
             # Update averaged advantage norm.
             rate = self.config["moving_average_sqd_adv_norm_update_rate"]
-            self._moving_average_sqd_adv_norm.add_(
-                rate * (adv_squared_mean - self._moving_average_sqd_adv_norm)
+            self._moving_average_sqd_adv_norm = (
+                rate * (adv_squared_mean.detach() - self._moving_average_sqd_adv_norm)
+                + self._moving_average_sqd_adv_norm
             )
```

Comment:
are we detaching everything now? :)

Reply:
no detach 
https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html

returns a new tensor, its not an inplace op.